code/modules/power/power.dm

```diff
@@ -199,6 +145,32 @@
 	powernet.remove_machine(src)
 	return 1
 
+// produces data for the power monitor
+/obj/machinery/power/proc/get_monitor_status_template()
+	return list( // Unique ID. If you're splitting a machine into multiple loads, eg: main and battery (see APCs) you could have "\ref[src]" and "\ref[src]_battery" as IDs
+		"ref" = "\ref[src]", // ref for use in Topic() calls
+		"name" = name,
+
+		"priority" = power_priority,
+		"demand" = 0,  //How much power is being requested. Override this proc and fill this in manually.
+
+		"isbattery" = FALSE, // If true, a charge meter will be displayed
+		"charging" = MONITOR_STATUS_BATTERY_STEADY,
+		"charge" = 100
+	)
+
+/obj/machinery/power/proc/change_priority(value, id)
+	if(id == "\ref[src]")
```

Comment:
Actually, it's not meant as a security check, but as a disambiguation check for machines that may need it, such as APCs where the power supply priority and the battery recharge priority are separate priorities:
```dm
/obj/machinery/power/apc/change_priority(value, id)
    . = ..()
    if(.)
        return

    if (id == "\ref[src]_b")
        power_recharge_priority = value
```
(Sidenote, APCs are also the only case where this was needed so far.)

Note how in the the power monitor's `ui_act()` we're locating using `params["ref"]`, which would always be the machine's `\ref[src]`, whereas `change_priority()` is called using `params["id"]`, which so long as it's _unique_ it can be anything. 

The way I went, the default implementation of `get_monitor_status()` uses `\ref[src]` for the ID and then the default implementation of `change_priority()` looks for that `\ref[src]`, while machines that need to expose extra aspects of themselves with a separate priority can do so by extending both `get_monitor_status()` and `change_priority()`. 
Also sidenote, `"\ref[src]_b"` could have been anything else (so long as it's unique), such as a more descriptive `"\ref[src]_battery"`, but seeing as this will be getting passed to the UI and back, and that APCs comprise the majority of the powernet, I figured I should keep it short.

Now going back to the original point: you're right, I should check the monitor and the machine are in the same powernet. Would updating the power monitor's `ui_act()` to `locate(params["ref"]) in connected_powernet.nodes` do the trick?